tests/support/util.tcl

```diff
@@ -685,21 +685,47 @@ proc process_is_alive pid {
     }
 }
 
+proc get_system_name {} {
+    return [string tolower [exec uname -s]]
+}
+
+proc get_proc_state {pid} {
+    if {[get_system_name] eq {sunos}} {
+        return "ps -o s= -p $pid"
+    } else {
+        return "ps -o state= -p $pid"
+    }
+}
```

Comment:
why not move `[exec {*}` into here?

Reply:
Sure was just using your suggested code.